src/Utilities/NewTimer.cpp

```diff
@@ -265,16 +265,20 @@ TimerManagerClass::print_flat(Communicate* comm)
   {
     #pragma omp master
     {
+      const int bufsize = 256;
+      char tmpout[bufsize];
       std::map<std::string,int>::iterator it(p.nameList.begin()), it_end(p.nameList.end());
       while(it != it_end)
       {
         int i=(*it).second;
         //if(callList[i]) //skip zeros
-        printf ("%-40s  %9.4f  %13ld  %16.9f  %12.6f TIMER\n"
+        snprintf (tmpout, bufsize, "%-40s  %9.4f  %13ld  %16.9f  %12.6f TIMER\n"
         , (*it).first.c_str()
         , p.timeList[i], p.callList[i]
         , p.timeList[i]/(static_cast<double>(p.callList[i])+std::numeric_limits<double>::epsilon())
         , p.timeList[i]/static_cast<double>(omp_get_max_threads()*comm->size()));
+        tmpout[bufsize-1] = '\0';
```

Comment:
I mean no need for this line since snprintf does this.

Reply:
Ah, you're right.  I was thinking of the _snprintf on Windows, which doesn't properly null-terminate if it overwrites the buffer.